Track develoment progress

zhenekq opened the pull request.

zhenekq committed:
created entity People with ID, using UUID

zhenekq committed:
created relation between "people" and "peopleupdate"

zhenekq committed:
big update, added repositories for Entities, created another relations between tables and entities.

zhenekq committed:
add liquibase migration for creating every table in database, and some fixings in project

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,43 @@
+
+@RestController
+@RequestMapping("/people")
+public class PeopleController {
```
JavaDoc

zhenekq replied:
solved

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @GetMapping
+    public List<People> getAll(@RequestParam(required = false) Set<Keyword> keywords) {
```
JavaDoc

zhenekq replied:
solved

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+
+@Entity
+@Table(name = "attributetypes")
+public class AttributeTypes {
```
JavaDoc

zhenekq replied:
solved

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+    @Column(name = "isidentifier")
+    private Boolean isIdentifier;
+
+    /*Is primary key for this table,
```
What is it? Where is PK?

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AttributeTypes that = (AttributeTypes) o;
+        return id.equals(that.id) && name.equals(that.name) && basicType.equals(that.basicType) && validation.equals(that.validation) && isIdentifier.equals(that.isIdentifier);
```
tooooo long
Add EqualsBuilder from Apache commons

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, name, basicType, validation, isIdentifier);
```
HashCodeBuilder

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+    @Override
+    public String toString() {
+        return "AttributeTypes{" +
+                "id=" + id +
```
ToStringBuilder

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/CommunicationHistory.java:
```diff
@@ -0,0 +1,101 @@
+
+@Entity
+@Table(name = "communication_history")
+public class CommunicationHistory {
```
JavaDoc

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/CommunicationHistory.java:
```diff
@@ -0,0 +1,101 @@
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CommunicationHistory that = (CommunicationHistory) o;
+        return id.equals(that.id) && createDate.equals(that.createDate) && updateDate.equals(that.updateDate) && comment.equals(that.comment) && human.equals(that.human);
```
the same as previous

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/People.java:
```diff
@@ -0,0 +1,138 @@
+        this.status = status;
+    }
+
+    public People(String id, Timestamp lastContact, String status, Set<PeopleUpdate> humanUpdates, Set<Keyword> humanKeywords, Set<CommunicationHistory> communicationHistory, Set<PeopleAttributes> humanAttributes) {
```
too long. column

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/People.java:
```diff
@@ -0,0 +1,138 @@
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
the same as previous

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/PeopleAttributes.java:
```diff
@@ -0,0 +1,100 @@
+
+@Entity
+@Table(name = "peopleattributes")
+public class PeopleAttributes {
```
JavaDoc

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/PeopleUpdate.java:
```diff
@@ -0,0 +1,114 @@
+    @Column(name = "updatedate")
+    private Timestamp updateDate;
+
+    @Type(type = "jsonb")
```
what db is used? if postgress this is enought. Example
    @Column(name = "revision", columnDefinition = "json")
    private String revision;

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/entity/PeopleUpdate.java:
```diff
@@ -0,0 +1,114 @@
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
The same as for previous

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/repository/AttributeTypesRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface AttributeTypesRepository extends JpaRepository<AttributeTypes, Integer> {
```
javadoc

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/repository/CommunicationHistoryRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface CommunicationHistoryRepository extends JpaRepository<CommunicationHistory, Integer> {
```
javadoc

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/repository/PeopleAttributesRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface PeopleAttributesRepository extends JpaRepository<PeopleAttributes, Integer> {
```
javadoc

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/repository/PeopleRepository.java:
```diff
@@ -0,0 +1,16 @@
+
+@Repository
+public interface PeopleRepository extends JpaRepository<People, String> {
+    /*List<People> findPeopleByHumanKeywords(Set<Keyword> keywords);
```
commented code? why?

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/repository/PeopleUpdateRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface PeopleUpdateRepository extends JpaRepository<PeopleUpdate, Integer> {
```
javadoc

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/service/PeopleService.java:
```diff
@@ -0,0 +1,17 @@
+
+public interface PeopleService {
+
+    List<People> getAllPeople();
```
javadoc for each method

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on pom.xml:
```diff
@@ -36,6 +36,15 @@
 		</dependency>
+		<dependency>
+			<groupId>com.vladmihalcea</groupId>
+			<artifactId>hibernate-types-52</artifactId>
```
what is it? why you need that?

zhenekq replied:
deleted, I need this for @Type for using json, at now, i found another solution to solve this problem

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @GetMapping
+    public List<People> getAll(@RequestParam(required = false) Set<Keyword> keywords) {
```
getAll shouldn't have keywords as params but should accept pagination params

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    @PostMapping
+    public void add(@RequestBody People people) {
```
Let's return ID

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,43 @@
+
+    @GetMapping("/{uuid}")
+    public People getHumanByUUID(@PathVariable String uuid) {
+        return peopleService.getHumanById(uuid);
```
rename to getHumanByID

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+import java.util.Objects;
+
+@Entity
+@Table(name = "attributetypes")
```
Why you need name here?

zhenekq replied:
i have an image of relationships between tables and their names, i just write name like in schema

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/AttributeTypes.java:
```diff
@@ -0,0 +1,105 @@
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "basictype")
```
you can omit the name for column annotation, it should used by convention

zhenekq replied:
i know, but i wanted to be confident in every field, and doing this

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/PeopleService.java:
```diff
@@ -0,0 +1,17 @@
+    List<People> getAllPeople();
+    People getHumanById(String uuid);
+    void addNewHuman(People people);
+    Set<People> findPeopleByKeywords(Set<Keyword> keywords);
```
let's use list or iterable as return type

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/impl/PeopleServiceImpl.java:
```diff
@@ -0,0 +1,49 @@
+
+    @Override
+    public Set<People> findPeopleByKeywords(Set<Keyword> keywords) {
+        return null;
```
1. return empty collection 
2. use nullable/notnull annotation in method signature

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/util/StringPrefixedSequenceIdGenerator.java:
```diff
@@ -0,0 +1,41 @@
+import java.io.Serializable;
+import java.util.Properties;
+
+public class StringPrefixedSequenceIdGenerator extends SequenceStyleGenerator {
```
Why you need IdGenerator?

zhenekq replied:
removed, just unnecessary class

zhenekq committed:
fixed all mistakes, by recommendations

zhenekq committed:
fixed relationships, types, foreign keys.

zhenekq committed:
fixed relationships, types, foreign keys.

zhenekq committed:
Merge branch 'complete-postinitial-settings' of https://github.com/zhenekq/HR-Automation into complete-postinitial-settings

zhenekq committed:
created relationship between people and people_merge_candidates

zhenekq committed:
add some swagger api documentation

zhenekq committed:
some fixes

zhenekq committed:
fixed tables column names. fixed problem to get information about user

zhenekq committed:
fixes

zhenekq committed:
documented and fixed working of Dtos.
Successful responses of get candidates, get paginated candidates, get candidates by id.
Successful responses of get keywords, get keywords by keyword

zhenekq committed:
filled table "communication_history" with values and added some fixes

zhenekq committed:
refactored dto structure(some fixes)

zhenekq committed:
migration from uses Set implementation, at now used List.
Development of communicationHistory logic

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/repository/CommunicationHistoryRepository.java:
```diff
@@ -5,6 +5,8 @@
+import java.util.List;
+
 /**
  * Repository for working with it in database
```
what is 'it'? working with CommunicationHistory

zhenekq replied:
solved

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,11 @@
+
+import java.util.List;
+
+public interface CommunicationHistoryService {
```
Any interface should have javadoc

zhenekq replied:
solved

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/service/impl/KeywordServiceImpl.java:
```diff
@@ -24,7 +23,7 @@ public KeywordServiceImpl(KeywordRepository repository) {
+        if (page == null) {
             return repository.findAll();
         }
         Pageable pageable = PageRequest.of(page - 1, AMOUNT_OF_RESULTS_ON_ONE_PAGE);
```
Amount should be a property, or even get parameter

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/util/impl/MappingDtoComponentConverterImpl.java:
```diff
@@ -40,9 +42,19 @@ public KeywordDto convertToKeywordDto(Keyword keyword) {
-        Set<KeywordDto> list = new HashSet<>();
+    public List<KeywordDto> convertToListKeywordDto(List<Keyword> keywords) {
+        List<KeywordDto> list = new ArrayList<>();
         keywords.forEach((el) -> list.add(convertToKeywordDto(el)));
```
there is map method in stream api used for change type

zhenekq replied:
I decided to use List everywhere

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/util/impl/MappingDtoComponentConverterImpl.java:
```diff
@@ -40,9 +42,19 @@ public KeywordDto convertToKeywordDto(Keyword keyword) {
+
+    @Override
+    public CommunicationHistoryDto convertToCommunicationHistoryDto(CommunicationHistory communicationHistory) {
+        return null;
```
why returning null?

zhenekq replied:
Method in progress, soon logic will be provided

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/service/impl/KeywordServiceImpl.java:
```diff
@@ -24,7 +23,7 @@ public KeywordServiceImpl(KeywordRepository repository) {
     @Override
     public List<Keyword> getAllKeywords(Integer page) {
-        if(page == null){
+        if (page == null) {
```
if page < 0 how it will work?

zhenekq replied:
validation in progress, fixed

AndrewRest commented on Dockerfile:
```diff
@@ -0,0 +1,4 @@
+FROM openjdk:8
+EXPOSE 8080
+ADD target/mifort-automation-api.jar mifort-automation-api.jar
```
does it work? are you really get jar without 0.0.1-SNAPSHOT

AndrewRest commented on pom.xml:
```diff
@@ -1,50 +1,86 @@
+    <groupId>by.mifort.automation.hr</groupId>
+    <artifactId>dev</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
+    <name>dev</name>
```
? Interesting name

zhenekq replied:
renamed :)

AndrewRest commented on pom.xml:
```diff
@@ -1,50 +1,86 @@
+    <name>dev</name>
+    <description>Hr-Automation-api</description>
+    <properties>
+        <maven.compiler.source>1.8</maven.compiler.source>
```
why different versions of java? 1.8 here and 16 below?

zhenekq replied:
fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleCommunicationHistoryController.java:
```diff
@@ -0,0 +1,29 @@
+    @GetMapping
+    public List<CommunicationHistory> getLastCommunication(
+            @RequestParam(required = false) Long page){
+        return service.findByOrderByCreateDateDesc();
```
page not used here?

zhenekq replied:
Fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,66 @@
+     */
+    @ApiOperation("Get all users by page, on every page 2 users")
+    @GetMapping
+    public List<PeopleDto> getAll(@RequestParam(required = false) Integer page) {
```
in one place you use integer for page in another Long. use integer everythere

akolonitsky-cloudbreak replied:
use pagination dto object

zhenekq replied:
Fixed

AndrewRest commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,66 @@
+    @ApiOperation("Get full information about user by id")
+    @GetMapping("/{id}")
+    public PeopleDto getHumanById(@PathVariable String id) {
+        PeopleDto human = peopleService.getHumanById(id);
```
why do you need human var here?

zhenekq replied:
Because you can use this variable in future

akolonitsky-cloudbreak commented on pom.xml:
```diff
@@ -1,50 +1,86 @@
+        <version>2.6.3</version>
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
+    <groupId>by.mifort.automation.hr</groupId>
```
com.mifort. ...

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleCommunicationHistoryController.java:
```diff
@@ -0,0 +1,29 @@
+import java.util.List;
+
+@RestController
+@RequestMapping("/candidates/history")
```
/candidates/{id}/history

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleCommunicationHistoryController.java:
```diff
@@ -0,0 +1,29 @@
+
+@RestController
+@RequestMapping("/candidates/history")
+public class PeopleCommunicationHistoryController {
```
People/Human/Candidates - please use only one name

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleCommunicationHistoryController.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    @GetMapping
+    public List<CommunicationHistory> getLastCommunication(
```
We should accept FilterDto object that inherited from PaginationDto where we have pageNum and pageSize params

zhenekq replied:
will read about it

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,66 @@
+     * @param people - candidate entity body
+     */
+    @ApiOperation("Create new user")
+    @PostMapping("/add")
```
no need 'add'

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleController.java:
```diff
@@ -0,0 +1,66 @@
+
+    /**
+     * POST request to create candidate
+     * @param people - candidate entity body
```
add return value description

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/PeopleKeywordsController.java:
```diff
@@ -0,0 +1,66 @@
+     * @param keyword identifier of keyword
+     * @return Keyword and connected candidates
+     * */
+
```
remove space

zhenekq replied:
fixed

zhenekq committed:
fixes, and some edit

zhenekq committed:
PaginationDto created, some fixes with it

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/CandidateController.java:
```diff
@@ -0,0 +1,70 @@
+    @ApiOperation("Get paginated candidates, with amount of them on one page")
+    @GetMapping
+    public List<CandidateDto> getAll(FilterDto filterDto,
+                                     @RequestParam(required = false) String keyword) {
```
keyword is not necessary, it should be part of filters

zhenekq replied:
I know, will be fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/CandidateController.java:
```diff
@@ -0,0 +1,70 @@
+    @ApiOperation("Get full information about user by id")
+    @GetMapping("/{id}")
+    public CandidateDto getHumanById(@PathVariable String id) {
+        return candidateService.getHumanById(id);
```
getHumanById rename to getById

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/CandidateController.java:
```diff
@@ -0,0 +1,70 @@
+    @ApiOperation("Create new user")
+    @PostMapping
+    public String add(@RequestBody Candidate candidate) {
+        candidateService.addNewHuman(candidate);
```
addNewHuman rename to create

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/CandidateController.java:
```diff
@@ -0,0 +1,70 @@
+     * @param candidate - candidate entity body
+     * @return Conversation of successful created candidate with id
+     */
+    @ApiOperation("Create new user")
```
user -> candidate

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/CandidateController.java:
```diff
@@ -0,0 +1,70 @@
+    @PostMapping
+    public String add(@RequestBody Candidate candidate) {
+        candidateService.addNewHuman(candidate);
+        return "Candidate successfully created: " + candidate.getId();
```
return just ID, without string message

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/dto/CommunicationHistoryDto.java:
```diff
@@ -0,0 +1,84 @@
+    private Timestamp createDate;
+    private Timestamp updateDate;
+    private String comment;
+    private CandidateDto human;
```
we don't need this field here

zhenekq replied:
human field you mean?

akolonitsky-cloudbreak replied:
yes

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/dto/KeywordDto.java:
```diff
@@ -0,0 +1,44 @@
+ * @see by.mifort.automation.hr.dev.entity.Keyword
+ */
+
+public class KeywordDto {
```
why you need dto for keyword it is just a string

zhenekq replied:
deleted

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/dto/PaginationDto.java:
```diff
@@ -0,0 +1,53 @@
+@JsonIgnoreProperties({"page", "amount"})
+public class PaginationDto {
+
+    private Integer page;
```
rename to pageNumber

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/dto/PaginationDto.java:
```diff
@@ -0,0 +1,53 @@
+public class PaginationDto {
+
+    private Integer page;
+    private Integer amount;
```
rename to pageSize

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/CandidateAttributes.java:
```diff
@@ -0,0 +1,94 @@
+    private Integer valueSource;
+
+    @ManyToOne
+    @JoinColumn(name = "user_id", nullable = false)
```
we should use camelcase

zhenekq replied:
have some problems with this case in postgreSQL, i wrote userId, but postgreSQL ignored it, and wanted smth else, i talked with Andrey, and he said to make this in database and here

akolonitsky-cloudbreak replied:
ok, leave todo here

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/CandidateAttributes.java:
```diff
@@ -0,0 +1,94 @@
+
+    @ManyToOne
+    @JoinColumn(name = "user_id", nullable = false)
+    private Candidate human;
```
rename to candidate

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/CandidateMergeCandidates.java:
```diff
@@ -0,0 +1,118 @@
+                    insertable = false,
+                    updatable = false
+            )
+    })
```
please explain this mapping

akolonitsky-cloudbreak replied:
?

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/Keyword.java:
```diff
@@ -0,0 +1,64 @@
+@Table(name = "keywords", schema = "public")
+public class Keyword {
+    @Id
+    private String id = UUID.randomUUID().toString();
```
why do we need initialization here?

zhenekq replied:
deleted

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/MergeCandidates.java:
```diff
@@ -0,0 +1,65 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/MergeCandidates.java:
```diff
@@ -0,0 +1,65 @@
+ */
+
+@Embeddable
+public class MergeCandidates implements Serializable {
```
why you need additional entity?

zhenekq replied:
I find solution with 2 primary keys in the Internet to solve this problem

akolonitsky-cloudbreak replied:
It should be something like here https://stackoverflow.com/questions/51650881/two-onetomany-sets-to-same-table-in-hibernate

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/repository/CandidateRepository.java:
```diff
@@ -0,0 +1,31 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/repository/CommunicationHistoryRepository.java:
```diff
@@ -0,0 +1,37 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/repository/CommunicationHistoryRepository.java:
```diff
@@ -0,0 +1,37 @@
+     * @param pageable value for pagination
+     * @return page of communication history with all candidates
+     * */
+    @Query("select h from CommunicationHistory h")
```
why we need query here?

zhenekq replied:
Without this, we have no pagination, and have error on runtime, after i added this, exception is gone

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/repository/KeywordRepository.java:
```diff
@@ -0,0 +1,26 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CandidateService.java:
```diff
@@ -0,0 +1,52 @@
+     * @return information about candidate
+     * @param id identifier of candidate
+     */
+    CandidateDto getHumanById(String id);
```
rename to getById

zhenekq replied:
rename

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CandidateService.java:
```diff
@@ -0,0 +1,52 @@
+     * @param filterDto dto with pagination, sortTypes
+     * @see FilterDto
+     */
+    List<CandidateDto> getAllPeople(FilterDto filterDto, String keyword);
```
rename to getAll

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CandidateService.java:
```diff
@@ -0,0 +1,52 @@
+     * Create new candidate in database
+     * @param candidate body of our candidate
+     */
+    void addNewHuman(Candidate candidate);
```
rename to create

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CandidateService.java:
```diff
@@ -0,0 +1,52 @@
+     * @return List of candidate by keyword(s)
+     * @param keywords keywords that connected with candidate
+     */
+    List<Candidate> findPeopleByKeywords(Set<Keyword> keywords);
```
rename to findByKeywords

zhenekq replied:
rename

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CandidateService.java:
```diff
@@ -0,0 +1,52 @@
+     * @return List of candidate by attribute(s)
+     * @param attributes attributes that connected with candidate
+     */
+    List<Candidate> findPeopleByHumanAttributes(Set<CandidateAttributes> attributes);
```
Rename to findByAttributes

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ * */
+
```
remove blank lines

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * Get history of candidate by id
+     * @param id candidate's identifier
+     */
+    List<CommunicationHistoryDto> getHistoryByCandidateId(String id);
```
rename to getByCandidateId

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto createHistoryWithCandidate(String id, CommunicationHistory history);
```
rename to create

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto updateHistoryWithCandidate(String id, CommunicationHistory history);
```
rename to update

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto archiveHistoryWithCandidate(String id, CommunicationHistory history);
```
rename to `archive`

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/KeywordService.java:
```diff
@@ -0,0 +1,24 @@
+     */
+    List<Keyword> getAllKeywords(Integer page);
+
+    List<Keyword> getAllKeywordsById(String id);
```
why we expect list?

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/impl/CandidateServiceImpl.java:
```diff
@@ -0,0 +1,74 @@
+        List<Candidate> candidateList;
+        Integer page = filterDto.getPage();
+        Integer amount = filterDto.getAmount();
+        if (page == null || amount == null || page <= 0 || amount <= 0) {
```
we should check it in controller

zhenekq replied:
Why? All validation and logic we need to put in Service Layer. I want to create in future interface with validation

akolonitsky-cloudbreak replied:
it is validation of input param and you use here Http specific objects `ResponseStatusException(HttpStatus.BAD_REQUEST`

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/util/MappingDtoComponentConverter.java:
```diff
@@ -0,0 +1,34 @@
+ * @version 1.0
+ * */
+
+public interface MappingDtoComponentConverter {
```
Can we use Spring Type Conversion?

zhenekq replied:
I discussed about this with Andrew and in conclusion, i will do it by myself

zhenekq committed:
refactored from human to candidate variables names

zhenekq committed:
other fixes with naming

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/dto/PaginationDto.java:
```diff
@@ -0,0 +1,53 @@
+ * @version 1.0
+ * */
+
+@JsonIgnoreProperties({"page", "amount"})
```
should be fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/Candidate.java:
```diff
@@ -0,0 +1,164 @@
+     * @see CandidateAttributes
+     */
+    @OneToMany(mappedBy = "candidate")
+    private List<CandidateAttributes> candidateAttributes;
```
let's rename it to `attributes`

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/Candidate.java:
```diff
@@ -0,0 +1,164 @@
+     * @see CandidateMergeCandidates
+     */
+    @OneToMany(mappedBy = "candidate")
+    private List<CandidateMergeCandidates> candidates;
```
rename to `mergeCandidates`

zhenekq replied:
renamed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/CandidateAttributes.java:
```diff
@@ -0,0 +1,102 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/CandidateMergeCandidates.java:
```diff
@@ -0,0 +1,118 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/entity/CandidateUpdate.java:
```diff
@@ -0,0 +1,103 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ * */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/repository/CandidateAttributesRepository.java:
```diff
@@ -0,0 +1,16 @@
+ * @author yauheni_vozny
+ * @version 1.0
+ */
+
```
remove blank line

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CandidateService.java:
```diff
@@ -0,0 +1,52 @@
+     * @param filterDto dto with pagination, sortTypes
+     * @see FilterDto
+     */
+    List<CandidateDto> getAll(FilterDto filterDto, String keyword);
```
keyword should be a list and it should be a part of filterDto

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto createByCandidateId(String id, CommunicationHistory history);
```
rename Id to candidateId

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto updateByCandidateId(String id, CommunicationHistory history);
```
rename Id to candidateId

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto archiveByCandidateId(String id, CommunicationHistory history);
```
rename Id to candidateId

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/CommunicationHistoryService.java:
```diff
@@ -0,0 +1,45 @@
+     * @param id candidate's identifier
+     * @param history body of history
+     */
+    CommunicationHistoryDto archiveByCandidateId(String id, CommunicationHistory history);
```
do we really need history object here?

zhenekq replied:
deleted

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/service/impl/CommunicationHistoryServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+    public CommunicationHistoryDto createByCandidateId(String id, CommunicationHistory history) {
+        Candidate candidate = new Candidate();
+        candidate.setId(id);
+        history.setCandidate(candidate);
```
remove candidate field from history object

zhenekq replied:
removed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/util/impl/MappingDtoComponentConverterImpl.java:
```diff
@@ -0,0 +1,60 @@
+        candidateDto.setId(candidate.getId());
+        candidateDto.setStatus(candidate.getStatus());
+        candidateDto.setLastContact(candidate.getLastContact());
+        if(candidate.getCandidateKeywords() != null)
```
please use code formatting, and curly braces for if-else statement

zhenekq replied:
fixed

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/util/impl/MappingDtoComponentConverterImpl.java:
```diff
@@ -0,0 +1,60 @@
+    @Override
+    public List<CommunicationHistoryDto> convertToListCommunicationHistoryDto(List<CommunicationHistory> communicationHistoryList) {
+        List<CommunicationHistoryDto> dtoList = new ArrayList<>();
+        communicationHistoryList.forEach(
```
use map method instead of forEach

zhenekq replied:
used

zhenekq committed:
clean upped code, some fixes

zhenekq committed:
uses map function

zhenekq committed:
some fixes with docker and started to develop business logic of CandidateAttributesService

zhenekq committed:
filled attributetypes table with values

zhenekq committed:
filled peopleattributes table with values

zhenekq committed:
make some basic logic with candidateAttributes(still in progress)

zhenekq committed:
created CandidateAttributeController and edited method for updating communicationHistory

zhenekq committed:
refactored project structure

zhenekq committed:
little fixes with project and relationships

zhenekq committed:
added new functionality and reviewed some parts of code to Stream API

zhenekq committed:
search by part of keyword

akolonitsky-cloudbreak commented on src/main/java/by/mifort/automation/hr/dev/controller/AttributeTypesController.java:
```diff
@@ -0,0 +1,86 @@
+ */
+
+@RestController
+@RequestMapping("/candidates/attributes")
```
change to `/attributestypes`

zhenekq replied:
fixed

zhenekq committed:
fixes

zhenekq committed:
added column isArchived for archiving attributeTypes and communication history

zhenekq committed:
rename isArchived to snake-case

zhenekq committed:
added functionality to archive communication history with candidate

zhenekq committed:
added functionality to get archived and not archived histories by candidate's identifier

zhenekq committed:
little fixes with archived histories

zhenekq committed:
making also get archived or not archived attribute types
